src/PresentationalComponents/Inventory/Inventory.js

```diff
@@ -68,11 +68,35 @@ const Inventory = ({
           {}
         )
       )?.data.data;
-      const issues = pathways.map((rec) => ({
-        id: `advisor:${rec.rule_id}`,
-        description: rec.description,
-      }));
-      return { issues, systems: selected };
+
+      const systems = (
+        await Get(
+          `${BASE_URL}/pathway/${encodeURI(pathway.slug)}/reports/`,
+          {},
+          {}
+        )
+      )?.data.rules;
+
+      let issues = [];
+      pathways.forEach((rec) => {
+        let filteredSystems = [];
+
+        systems[rec.rule_id].forEach((system) => {
+          if (selected.includes(system)) {
```

Comment:
While trying to see if the `issues` construction could be written differently, I noticed the `selected` and how the `remediationDataProvider` is defined and I believe there is a chance that `selected` becomes stale when React does its re-rendering. And it has to redefine the function on each render. Maybe this could be refactored (in another PR) to use `useMemo` and pass in the `selected` instead to not have it as a dependency for the memoized function.

Reply:
Yea that could be definitely done, but as you say in different PR and probably under another task as performance enhancement. But this PR and task corresponds only for a fix of functionality